pontoon/sync/vcs/models.py

```diff
@@ -431,6 +438,69 @@ def resources_for_path(self, path):
                     yield os.path.join(root, filename)
 
 
+class VCSConfiguration(object):
+    """
+    Container for the project configuration, provided by the optional
+    configuration file.
+
+    For more information, see:
+    https://moz-l10n-config.readthedocs.io/en/latest/fileformat.html.
+    """
+
+    def __init__(self, vcs_project):
+        self.vcs_project = vcs_project
+        self.configuration_file = vcs_project.db_project.configuration_file
+
+    @cached_property
+    def configs(self):
+        """Return parsed project configuration file."""
+        path = os.path.join(
+            self.vcs_project.db_project.source_repository.checkout_path,
+            self.configuration_file
+        )
+        parser = TOMLParser()
+        return parser.parse(path).configs
+
+    @cached_property
+    def paths(self):
+        """Return paths specified in the configurations."""
+        return [
+            paths
+            for configuration in self.configs
+            for paths in configuration.paths
+        ]
+
+    def locale_paths(self, locale):
+        """
+        Return a list of reference paths that are either not limited to a
+        subset of locales or the subset includes the given locale.
+        """
+        return [
+            paths['reference']
+            for paths in self.paths
+            if 'locales' not in paths or locale.code in paths['locales']
+        ]
+
+    def locale_resources(self, locale):
+        """
+        Return a list of Resource instances, which need to be enabled for the
+        given locale.
+        """
+        resources = []
+
+        for resource in self.vcs_project.db_project.resources.all():
+            absolute_resource_path = os.path.join(
+                self.vcs_project.source_directory_path,
+                resource.path,
+            )
+
+            for path in self.locale_paths(locale):
+                if path.match(absolute_resource_path) is not None:
+                    resources.append(resource)
```

Comment:
Pretty sure this can create false-positives if `locale` and `l10n_base` aren't set in the matching environment.